### DISCUSSION TITLE
BlueRetro for Dreamcast cannot connect or pair

### DISCUSSION THREAD

**Discussion by HornheaDD** [darthcloud/BlueRetro#177] [Q&A] [answered]
Hi there, I am fairly certain Im missing something, but I followed a tutorial on Youtube to pair an xbox/ps4/etc controller to my dreamcast with an ESP32 board.  I got 3 in a pack from amazon. After following the step (pretty sure to the letter) I am stuck at the first time pairing with a console. 

The controllers Ive tried are an Xbox One (version 2, bluetooth) and 8BitDo SN30 pad. None of them paired.  When I try to connect to Blueretro.io to see if anything happens, I get this error:  

Requesting Bluetooth Device...
Connecting to GATT Server...
Getting BlueRetro Service...
> Bluetooth Device disconnected
Argh! NetworkError: GATT Server is disconnected. Cannot retrieve services. (Re)connect first with `device.gatt.connect`.

Im not exactly sure what that means. The flashing of the board worked just fine, the only difference to the video is that the board (I tried two of them) did not need to be put into a different mode by holding the Boot button for 2 seconds. They were both written to successully. 

Again, Im sure I am just missing something simple, but I'm not sure what it could be.   

The only other bit of info I can provide is regarding the Dreamcast controller I used.  In the video, the controller used has 5 wires in the cable. Mine does not. It's an official Sega Dreamcast controller. But I determined that the ground "wire" is part of the shielding on the cable, it looks like sort of a wire mesh on the cable. Using a multimeter I was able to determine that the 'shield' was actually the ground. 

Please help, I'd love to try and get this working.

**Comment by darthcloud** [comment]
Moved to main repo discussion forum.

Take good pictures of your install were I can see all the connection well.

**Reply by HornheaDD** [reply]
Sure thing. Sorry, I should have posted this in discussion, not issues. My apologies.

Here are the pics. I assume you were asking about the solder connections. I haven't installed the board into the DC yet, Im doing the "test" phase that Yakara talks about, connecting it to a Dreamcast controller cable first. I had a spare DC controller to sac, so I just used it.

Here are the pics: https://imgur.com/a/8ujAPGv

**Reply by darthcloud** [reply]
Remove that white styrofoam, its probably a static nightmare.

Other than that make sure you read 5V between your black and blue wire.

Triple check each wire with multimeter continuity check mode to be sure each wire is what you think they are.

**Reply by HornheaDD** [reply]
Yeah the foam is just there for the pic, I took it off right away, on both boards I tried.   Gimme SOME credit man! lol  But I hear ya, when I was flashing/soldering/testing, no foam was near the boards.

I'm reading 4.82v between power and ground. I did follow the directions on the vid to determine what lines were what. I thought it was weird my controller cord only had 4 lines in it, but when I opened the controller housing, I saw 5 wires on the molex connector. Thats what made me test to see if the wired mesh was the ground.  I also tried connecting to controller port 2 in case there was something wrong with port one causing it to have lower voltage. It also reads 4.82v.

**Comment by darthcloud** [comment]
Leaving the DC plug unconnected connect it to USB and try get the log:
https://github.com/darthcloud/BlueRetro/wiki/Getting-BlueRetro-debug-logs-via-Serial-port-Windows-10

Copy paste into a txt file and upload in retry via drap & drop.

**Comment by HornheaDD** [comment, marked as answer]
I believe I followed those steps you linked to correctly - please let me know if I missed something. If the attachment doesnt appear I put a link to Rentry.  https://rentry.co/smbxb

[2-20-22-Log.txt](https://github.com/darthcloud/BlueRetro/files/8105331/2-20-22-Log.txt)

**Reply by darthcloud** [reply]
Your esp32 is flashed and work properly.

Maybe somehow the internal pullup are not strong enough with your particular boards and input 2 3 & 4 make the adapter crash in loop.

Try this to stabilize those input:
If not wiring 2nd port plug, make sure to connect IO3 (P2_D0) & IO5 (P2_D1) to 3.3V
If not wiring 3rd port plug, make sure to connect IO18 (P3_D0) & IO23 (P3_D1) to 3.3V
If not wiring 4th port plug, make sure to connect IO26 (P4_D0) & IO27 (P4_D1) to 3.3V

**Reply by HornheaDD** [reply]
Well that's good to know at least.   I'm sorry if this sounds ignorant, but for this part that you said: 

_If not wiring 2nd port plug, make sure to connect IO3 (P2_D0) & IO5 (P2_D1) to 3.3V_
_If not wiring 3rd port plug, make sure to connect IO18 (P3_D0) & IO23 (P3_D1) to 3.3V_
_If not wiring 4th port plug, make sure to connect IO26 (P4_D0) & IO27 (P4_D1) to 3.3V_

Do you mean **bridge** wires from those points to the 3.3v point on the board?

**Reply by darthcloud** [reply]
Yes

**Reply by HornheaDD** [reply]
OK. I will try that and report back in a bit.

**Comment by HornheaDD** [comment]
OK, looks like that was the issue. I was able to successfully pair the XBoxOne S controller.  Only thing is, I found some small issues with the mapping.

For the Source, I selected the Xbox controller, Destination is Dreamcast.   For some reason, the Start button if mapped to the "Menu" (xbox's start button) - it would not work. I changed the mapping to use the X button as start, and that worked fine. That is a very very minor issue and I couldn't care less.  Now the only issue is A. my ESP32 looks hideous lol, and B. this basically disables the ability to use 4 bluetooth controllers.  Not that it really will affect me as I dont usually have 4 player Dreamcast parties, but that could present a problem for others.

Now I just need to really clean up the wiring on this one and then install it internally. Thank you very much for the help @darthcloud !!

Thanks for all your hard work!

**Reply by darthcloud** [reply]
You need to update the xbox controller firmware with https://www.microsoft.com/en-us/p/xbox-accessories/9nblggh30xj3

To fix mapping. Glad it work, enjoy!

**Reply by darthcloud** [reply]
If you wire the 4 port to the dreamcast you wont have issues.

**Reply by HornheaDD** [reply]
Thanks for the help. I actually did wire up the extra ESP32 board I had internally. I was following Yakara's tutorial with leaving port 4 unconnected, but I ran into the same pairing issue.  Then I figured just for shits and giggles, lets see if wiring the 4th controller would fix it, and it did.  So it looks like it's all good. I still need to install a cut off switch so I can go back and forth between the two,

As for the Xbox controller, I tried updating the firmware, but it seems I'm already on the most recent one for this controller. 3.1.1221.0.  But thats ok, like I said the start/x button is not a big problem for me. :)

**Reply by darthcloud** [reply]
if you dont want to use the 4th port pins, just connect them to 3.3V.

This is a old version version I got:
Xbox One S: 4.8.1923.0
Adaptive: 4.5.1680.0
Series X|S: 5.9.2709.0

Maybe your xbox app need update?

**Reply by HornheaDD** [reply]
Hm, ok I will check. I connected it to the Xbox Accessories app and it didnt tell me there was an update. I will try it with my Xbox console in a bit,

**Reply by HornheaDD** [reply]
So strange. you're right, I was able to update the controller on the Xbox One console.  And now it works 100% :)
